From a50018952a15cbd55bdbde24656f61f632335eac Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Mon, 17 Jun 2013 10:38:30 +0900
Subject: [PATCH 494/715] drm: rcar-du: Take offsets into account in the
 mode_set_base handler

Store the x and y offsets passed to the mode_set_base handler in the
plane structure. The offsets will be programmed to the hardware when
updating the plane base address.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
(cherry picked from commit a990c8b7448c126876e90ab8593705b9f27fea6d)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 08a69db..01e9d24 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -445,6 +445,11 @@ static void rcar_du_crtc_mode_commit(struct drm_crtc *crtc)
 static int rcar_du_crtc_mode_set_base(struct drm_crtc *crtc, int x, int y,
 				      struct drm_framebuffer *old_fb)
 {
+	struct rcar_du_crtc *rcrtc = to_rcar_crtc(crtc);
+
+	rcrtc->plane->src_x = x;
+	rcrtc->plane->src_y = y;
+
 	rcar_du_crtc_update_base(to_rcar_crtc(crtc));
 
 	return 0;
-- 
1.7.10.4

